app/v3/retrieveBiss/RetrieveBISSConnector.scala

```diff
@@ -35,28 +36,21 @@ class RetrieveBISSConnector @Inject() (val http: HttpClient, val appConfig: AppC
       request: RetrieveBISSRequestData)(implicit hc: HeaderCarrier, correlationId: String): Future[DownstreamOutcome[RetrieveBISSResponse]] = {
 
     import request._
-    val incomeSourceType = typeOfBusiness.toIncomeSourceType
+    val incomeSourceType: IncomeSourceType = typeOfBusiness.toIncomeSourceType(taxYear.year)
 
     request match {
       case def1: Def1_RetrieveBISSRequestData =>
         import def1._
-        val (downstreamUri, queryParam) =
-          if (taxYear.useTaxYearSpecificApi) {
-            (
-              TaxYearSpecificIfsUri[Def1_RetrieveBISSResponse](
-                s"income-tax/income-sources/${taxYear.asTysDownstream}/$nino/$businessId/$incomeSourceType/biss"),
-              Nil)
-          } else {
-            (
-              IfsUri[Def1_RetrieveBISSResponse](s"income-tax/income-sources/nino/$nino/$incomeSourceType/${taxYear.asDownstream}/biss"),
-              List("incomeSourceId" -> s"$businessId")
-            )
-          }
-
-        val response = get(downstreamUri, queryParam)
+
+        val uriBuilder: RetrieveBISSDownstreamUriBuilder[Def1_RetrieveBISSResponse] =
+          RetrieveBISSDownstreamUriBuilder.downstreamUriFor(taxYear)
+
+        val (downstreamUri, queryParams): (DownstreamUri[Def1_RetrieveBISSResponse], Seq[(String, String)]) =
+          uriBuilder.buildUri(nino, businessId, incomeSourceType, taxYear)
+
+        val response: Future[DownstreamOutcome[Def1_RetrieveBISSResponse]] = get(downstreamUri, queryParams)
```

Comment:
Is there a reason this needs to be defined as a val if it is only being returned?

Reply:
This was previously in place. I left in place as it shows that it works with this subtype of RetrieveBISSResponse (what the connector returns) which is Def1_RetrieveBISSResponse. So it is sort of for visibility and clarity. Let me know if you still want me to remove, thanks.